lib/config/defaults.js

```diff
@@ -704,8 +704,11 @@ const applyOutputDefaults = (
 			"type" in library
 				? library.name
 				: /** @type {LibraryName=} */ (library);
+		const excludePlaceholders = ["[name]"];
 		if (Array.isArray(libraryName)) {
-			return libraryName.join(".");
+			return libraryName
+				.filter(item => !excludePlaceholders.includes(item))
```

Comment:
you probably want to filter any placeholder

Reply:
Some placeholders are invalid in `library` configuration, eg [path].

Reply:
Could we only filter some placeholders that are valid for `library` configuration?

Reply:
I think you can use regexp `/^\[.*\]$/`

Reply:
an edge case, if someone want to use `[invalidPlaceholder]` as the string. The regexp `/^\[.*\]$/` will filter it.

Reply:
Hmm, i think the edge case above is weird, maybe we don't worry about it?

Reply:
We can add this to docs and avoid extra checks

Reply:
Well, the pr can be closed.